src_c/geometry.c

```diff
@@ -7,6 +7,108 @@
 #endif /* ~__AVX2__ */
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
+static PyObject *
+pg_raycast(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *startpoint;
+    PyObject *endpoint = NULL;
+    PyObject **farr;
+    PyObject **collisions;
+    Py_ssize_t loop;
+    double angle;
+    double max_dist;
+    double target_pos[2] = {0, 0};
```

Comment:
you don't need to specify the array size if you already provide the array contents like that.